### ISSUE TITLE
Heapster doesn't retry connecting to influxdb after a failure

### ISSUE THREAD

**Issue by bertrand-quenin** [kubernetes/heapster#898]
Hi,

It seems that heapster is not retrying to reconnect to influxdb after it failed once.

I'm using the following container images:
```
kubernetes/heapster:canary
kubernetes/heapster_influxdb:v0.6
kubernetes/heapster_grafana:v2.6.0
```

Basically, I'm starting all the pods at once on my environment. Sometimes, the DNS takes time to start, or influxdb starts after heapster, etc. It's basically a "start storm".

When heapster starts first (I assume before the DNS addon), I get this logs:
```
I0121 14:29:47.302511       1 heapster.go:61] /heapster --source=kubernetes:http://10.128.11.158:8080?inClusterConfig=false&auth=&insecure=true --sink=influxdb:http://influxdb:8086
I0121 14:29:47.302776       1 heapster.go:62] Heapster version 0.19.1
I0121 14:29:47.302933       1 kube_factory.go:172] Using Kubernetes client with master "http://10.128.11.158:8080" and version "v1"
I0121 14:29:47.310437       1 kube_factory.go:173] Using kubelet port 10255
E0121 14:30:07.330337       1 manager.go:299] encountered following errors while setting up sinks - issues while creating an InfluxDB sink: failed to ping InfluxDB server at "influxdb:8086" - Get http://influxdb:8086/ping: dial tcp: lookup influxdb: no such host, will retry on use
I0121 14:30:07.436818       1 heapster.go:72] Starting heapster on port 8082
```

It's fine, however, heapster never retries connecting, despite the "will retry on use" message.

If I restart heapster after the dns addons & influxdb are started, everything is working like a charm.

Any idea?

Thanks,
BQ.

**Comment by geoah** [comment]
Having the same issue, currently on canary.

**Comment by fgrzadkowski** [comment]
This will be fixed in new release (v0.20)
